### ISSUE TITLE
Helm install gives 404 Not Found

### ISSUE THREAD

**Issue by sab24** [ContainerSolutions/trow#169]
Hi, I've been trying to get trow running on a bare-metal Kubernetes cluster, but I seem to get stuck at receiving a 404 page from Ingress. 

I installed trow with the following command:
``` helm install trow -f values.yaml trow/trow ```

values.yaml
```
ingress:
    enabled: true
    annotations:
        cert-manager.io/issuer: letsencrypt-prod
        kubernetes.io/ingress.class: nginx
    tls:
    - hosts:
      - <trow.domain.tld>
      secretName: trow-ingress-tls
trow:
    domain: <trow.domain.tld>
    user: <user>
    password: <pass>
```

The certificate is generated, ingress is up:
```
kubectl describe ingress trow
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
Name:             trow
Namespace:        default
Address:          172.28.2.1
Default backend:  default-http-backend:80 (<error: endpoints "default-http-backend" not found>)
TLS:
  trow-ingress-tls terminates <trow.domain.tld>
Rules:
  Host               Path  Backends
  ----               ----  --------
  <trow.domain.tld>
                     /*   trow:8000   192.168.213.56:8000)
Annotations:         cert-manager.io/issuer: letsencrypt-prod
                     kubernetes.io/ingress.class: nginx
                     meta.helm.sh/release-name: trow
                     meta.helm.sh/release-namespace: default
Events:
  Type    Reason             Age    From                      Message
  ----    ------             ----   ----                      -------
  Normal  CREATE             6m35s  nginx-ingress-controller  Ingress default/trow
  Normal  CreateCertificate  6m35s  cert-manager              Successfully created Certificate "trow-ingress-tls"
  Normal  UPDATE             5m36s  nginx-ingress-controller  Ingress default/trow
```

```
curl -vvv 192.168.213.56:8000 -H "Host: <trow.domain.tld>"
*   Trying 192.168.213.56:8000...
* Connected to 192.168.213.56 (192.168.213.56) port 8000 (#0)
> GET / HTTP/1.1
> Host: <trow.domain.tld>
> User-Agent: curl/7.72.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Connection: close
< Content-Type: text/html; charset=utf-8
< Server: Rocket
< Docker-Distribution-API-Version: registry/2.0
< Content-Length: 89
< Date: Thu, 03 Sep 2020 10:22:51 GMT
<
<!DOCTYPE html><html><body>
<h1>Welcome to Trow, the cluster registry</h1>
* Closing connection 0
</body></html>
```

```
curl -vvv https://<trow.domain.tld>
*   Trying 172.28.2.1:443...
* Connected to <trow.domain.tld> (172.28.2.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=<trow.domain.tld>
*  start date: Sep  2 06:35:09 2020 GMT
*  expire date: Dec  1 06:35:09 2020 GMT
*  subjectAltName: host "<trow.domain.tld>" matched cert's "<trow.domain.tld>"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x556f313b09f0)
> GET / HTTP/2
> Host: <trow.domain.tld>
> user-agent: curl/7.72.0
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 404
< server: nginx/1.19.2
< date: Thu, 03 Sep 2020 10:28:26 GMT
< content-type: text/html
< content-length: 153
< strict-transport-security: max-age=15724800; includeSubDomains
<
<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.19.2</center>
</body>
</html>
* Connection #0 to host <trow.domain.tld> left intact
```

```
curl -vvv https://172.28.2.1 -H "host: <trow.domain.tld>"
*   Trying 172.28.2.1:443...
* Connected to 172.28.2.1 (172.28.2.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (OUT), TLS alert, unknown CA (560):
* SSL certificate problem: unable to get local issuer certificate
* Closing connection 0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
* Could not resolve host: =H
* Closing connection 1
curl: (6) Could not resolve host: =H
* Closing connection -1
curl: (3) URL using bad/illegal format or missing URL
```

**Comment by sab24** [comment]
```helm get all trow``` gives:
```
NAME: trow
LAST DEPLOYED: Thu Sep  3 12:18:25 2020
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
USER-SUPPLIED VALUES:
ingress:
  annotations:
    cert-manager.io/issuer: letsencrypt-prod
    kubernetes.io/ingress.class: nginx
  enabled: true
  tls:
  - hosts:
    - <trow.domain.tls>
    secretName: trow-ingress-tls
trow:
  domain: <trow.domain.tls>
  password: <pass>
  user: <user>

COMPUTED VALUES:
affinity: {}
fullnameOverride: ""
image:
  pullPolicy: Always
  repository: containersol/trow
imagePullSecrets: []
ingress:
  annotations:
    cert-manager.io/issuer: letsencrypt-prod
    kubernetes.io/ingress.class: nginx
  enabled: true
  gke: false
  hosts:
  - host: null
    paths:
    - /*
  tls:
  - hosts:
    - <trow.domain.tls>
    secretName: trow-ingress-tls
nameOverride: ""
nodeSelector: {}
replicaCount: 1
resources: {}
service:
  port: 8000
  type: NodePort
tolerations: []
trow:
  domain: <trow.domain.tls>
  password: <pass>
  user: <user>
  validatingWebhooks:
    enabled: false
volumeClaim:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 20Gi

HOOKS:
MANIFEST:
---
# Source: trow/templates/secret.yaml
apiVersion: v1
data:
  pass: <pass>
kind: Secret
metadata:
  name: trow-password
---
# Source: trow/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: trow
  labels:
    helm.sh/chart: trow-0.3.0
    app.kubernetes.io/name: trow
    app.kubernetes.io/instance: trow
    app.kubernetes.io/version: "0.3.0"
    app.kubernetes.io/managed-by: Helm
spec:
  type: NodePort
  ports:
    - port: 8000
      targetPort: http
  selector:
    app.kubernetes.io/name: trow
    app.kubernetes.io/instance: trow
---
# Source: trow/templates/statefulset.yaml
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: trow
  labels:
    helm.sh/chart: trow-0.3.0
    app.kubernetes.io/name: trow
    app.kubernetes.io/instance: trow
    app.kubernetes.io/version: "0.3.0"
    app.kubernetes.io/managed-by: Helm
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: trow
      app.kubernetes.io/instance: trow
  serviceName: trow
  template:
    metadata:
      labels:
        app.kubernetes.io/name: trow
        app.kubernetes.io/instance: trow
    spec:
      containers:
      - name: trow-pod
        image: "containersol/trow:0.3.0"
        imagePullPolicy: Always
        args:
          - "--no-tls"
          - "-n"
          - "<trow.domain.tls>"
          - "-u"
          - "<user>"
          - "--password-file"
          - "/etc/trow/pass"
        ports:
        - name: http
          containerPort: 8000
        volumeMounts:
        - name: data-vol
          mountPath: /data
        - name: trow-pass
          mountPath: /etc/trow
          readOnly: true
      volumes:
        - name: trow-pass
          secret:
            secretName: trow-password
            items:
              - key: pass
                path: pass
      securityContext:
        runAsUser: 333333
        runAsGroup: 333333
        fsGroup: 333333
  volumeClaimTemplates:
    - metadata:
        name: data-vol
      spec:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 20Gi
---
# Source: trow/templates/ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: trow
  labels:
    helm.sh/chart: trow-0.3.0
    app.kubernetes.io/name: trow
    app.kubernetes.io/instance: trow
    app.kubernetes.io/version: "0.3.0"
    app.kubernetes.io/managed-by: Helm
  annotations:
    cert-manager.io/issuer: letsencrypt-prod
    kubernetes.io/ingress.class: nginx
spec:
  tls:
    - hosts:
        - "<trow.domain.tls>"
      secretName: trow-ingress-tls
  rules:
    - host: <trow.domain.tls>
      http:
        paths:
          - path: /*
            backend:
              serviceName: trow
              servicePort: 8000

NOTES:
Welcome to Trow:

Setup your DNS to point at Trow

If you are running Trow on http, there will need to be a TLS ingress in front of it as Trow needs to be run over TLS.
Please see the examples found on the docs at https://github.com/ContainerSolutions/trow/blob/master/docs/HELM_INSTALL.md

To test you can run and login with the credentials for <user>

$ docker login https://<trow.domain.tls>


Please Contribute To the Trow Project at https://github.com/ContainerSolutions/trow/

**Comment by amouat** [comment]
Thanks for the report. 

From looking at your output it doesn't seem than the ingress is forwarding to the pod correctly. I think there may be a mistake in the chart somewhere. I'll try to recreate this.

**Comment by amouat** [comment]
Great. I did see the same thing and wonder, apologies for not posting it earlier. I'll get the chart updated.

Thanks a lot for the report.

**Comment by sab24** [comment]
Great, but when working on that could you add the annotation for ingress:
```nginx.ingress.kubernetes.io/proxy-body-size: "0"```
I think that can be the standard for the helm chart no?

**Comment by Spazzy757** [comment]
Hi and thanks for repoting this issue!
> Great, but when working on that could you add the annotation for ingress:
> `nginx.ingress.kubernetes.io/proxy-body-size: "0"`
> I think that can be the standard for the helm chart no?
I think the idea is that we are refraining from doing any tech-specific suggestions as this annotation is only for nginx ingress and not for any of the other ingresses that are available for use (traefik, envoy, etc)

**Comment by sab24** [comment]
> I think the idea is that we are refraining from doing any tech-specific suggestions as this annotation is only for nginx ingress and not for any of the other ingresses that are available for use (traefik, envoy, etc)

Yeah, I agree with not generating side effects of installing this package. Good that you have it mentioned in the manual a few times so it is easy to solve. I think there is not more that can be done than that.

**Comment by amouat** [comment]
Fix merged for `/*` issue in #180.

I'm going to close this, but I think we should do some more work to make sure the correction ingress annotations are added automatically or strongly suggested. 

Thanks for the report @sab24 .
